docs/GenericChart.md

```diff
@@ -0,0 +1,428 @@
+# Generic Chart
+
+[Introduction](#introduction)\
+[Generic Chart metadata](#generic-chart-metadata)\
+[Chart Config Jsonata](#chart-config-jsonata)\
+[GenericChart component logic](#genericchart-component-logic)\
+[Generic Chart Table](#generic-chart-table)
+
+## Introducution
+
+HINT has a `GenericChart` vue component in the front end. This is currently only used to display Input Time Series charts - 
+however it has been designed to generically accept configuration, chart metadata, and chart data and to display the 
+resultant chart using the [Plotly](https://plotly.com/javascript/) library. 
+
+`GenericChart` is intended to be rather self-contained, for example, including its own filter and data source components. 
+However it does make use of the store to handle chart data and metadata. This is to avoid the chart needing to re-fetch 
+data when it is reloaded.  
+
+Input Time Series, the sole current implementation for Generic Chart, has custom datasets which must be fetched from 
+hintr - the url to use is included in the dataset config. However, the chart config could be extended to allow generic 
+chart visualisations of the standard input and output datasets in the store.
+
+There is a distinction between *data source* and *dataset* in Generic Chart configuration. A *dataset* refers to some data, 
+with the associated url from which to fetch it and an id by which to refer to it. A *data source* configures how data
+will fill a role in a particular Generic Chart configuration. For example, a chart might configure two data sources, which 
+will provide values to the X and Y axes separately. In the Input Time Series, we have only one data source, 'Data source'.
+
+A data source is fulfilled by a dataset, and configuration defines which datasets should be available for which data sources. In 
+Input Time Series, both the ANC and ART datasets are available for 'Data source', and the user selects which to use 
+from the data source control. 
+
+This diagram shows the main constituents and data flow involved in showing Generic Chart:
+![Diagram of Generic Chart](GenericChart.png "Diagram of Generic Chart")
```

Comment:
If you ever need to update this diagram could you try to hide the spellcheck underlining? It's a bit distracting. Would also help to share the source document and/or post a link to it here for future reference.